src/json-ld.net/Util/JSONUtils.cs

```diff
@@ -147,17 +143,29 @@ public static string ToString(JToken obj)
         /// 	</exception>
         /// <exception cref="System.IO.IOException">If there was an error resolving the resource.
         /// 	</exception>
-        public static JToken FromURL(Uri url)
+        public static async Task<JToken> FromURLAsync(Uri url)
         {
-#if !PORTABLE && !IS_CORECLR
-            HttpWebRequest req = (HttpWebRequest)HttpWebRequest.Create(url);
-            req.Accept = AcceptHeader;
-            WebResponse resp = req.GetResponse();
-            Stream stream = resp.GetResponseStream();
-            return FromInputStream(stream);
-#else
-            throw new PlatformNotSupportedException();
-#endif
+            HttpResponseMessage httpResponseMessage = null;
+            int redirects = 0;
+
+            // Manually follow redirects because .NET Core refuses to auto-follow HTTPS->HTTP redirects.
```

Comment:
seems like this is repeated from above?

Reply:
Yeah--mostly. The "what will the code review say" angel on my shoulder totally predicted this. :)

I struggled for a while and then left both copies in because:
1. I need the redirect state from one of them, but not the other.
1. One of them is responsible for surfacing required JsonLdExceptions, and the other one isn't.

I didn't want to add a whole class for this single method, but I think I'm going to have to. And I agree that it's the right thing to do.

Reply:
Done, and I'm actually very happy with how it turned out.